.jenkins/Jenkinsfile

```diff
@@ -11,762 +11,518 @@
-                  }
-                  attempt = attempt + 1
-                  compression = build job: "${env.JOB_NAME}-test-compression", propagate: false
-                  if (compression.result != 'FAILURE') break
-              }
-              if (compression.result != 'SUCCESS') unstable('compression failures')
-              if (compression.result == 'FAILURE') currentBuild.result='FAILURE'
-            }
-          }
-          post {
-            always {
-                warnError('missing test xml files') {
-                    script {
-                        copyTestResults('test-compression', compression.getNumber())
-                    }
-                }
-            }
-          }
-        }
-        stage('oa') {
-          steps {
-            script {
-                def attempt = 1
-                while (attempt <=2) {
-                  if (attempt > 1) {
-                    sleep(60 * attempt)
-                  }
-                  attempt = attempt + 1
-                  oa = build job: "${env.JOB_NAME}-test-oa", propagate: false
-                  if (oa.result != 'FAILURE') break
-              }
-              if (oa.result != 'SUCCESS') unstable('oa failures')
-              if (oa.result == 'FAILURE') currentBuild.result='FAILURE'
-            }
-          }
-          post {
-            always {
-                warnError('missing test xml files') {
-                    script {
-                        copyTestResults('test-oa', oa.getNumber())
-                    }
-                }
-            }
-          }
-        }
-        stage('system-keyspace-directory') {
-          steps {
-            script {
-                def attempt = 1
-                while (attempt <=2) {
-                  if (attempt > 1) {
-                    sleep(60 * attempt)
-                  }
-                  attempt = attempt + 1
-                  system_keyspace_directory = build job: "${env.JOB_NAME}-test-system-keyspace-directory", propagate: false
-                  if (system_keyspace_directory.result != 'FAILURE') break
-              }
-              if (system_keyspace_directory.result != 'SUCCESS') unstable('system-keyspace-directory failures')
-              if (system_keyspace_directory.result == 'FAILURE') currentBuild.result='FAILURE'
-            }
-          }
-          post {
-            always {
-                warnError('missing test xml files') {
-                    script {
-                        copyTestResults('test-system-keyspace-directory', system_keyspace_directory.getNumber())
-                    }
-                }
-            }
-          }
+    stage('Summary') {
+      steps {
+        summateStatus()
+      }
+    }
+  }
+  post {
+    always {
+      generateTestReports()
+      sendNotifications()
+    }
+  }
+}
+
+///////////////////////////
+//// scripting support ////
+///////////////////////////
+
+def jdksSupported() { return ["11", "17"] }
```

Comment:
I don't think this should be hardcoded, but fetched from the `build.xml`. We already have the same in properties there:
```
    <property name="java.default" value="11" />
    <property name="java.supported" value="11,17" />
```

We can use the ant cli tool to print it or create a dedicated goal for it e.g. `ant print-property`

Reply:
We wanted to take this approach, but this function needs to work outside of the context of a a current (git cloned) workspace. 

What you describe is provided by `javaVersionsSupported()` (and `javaVersionsSupported()`), but we have forgotten to use them as intended: the build should fail if the hardcoded values don't match those from build.xml

Reply:
> the build should fail if the hardcoded values don't match those from build.xml

given recent changes
- turning tasks() into a method instead of global variables, and
- changing the 'jdk' parameter to be "what jdks to restrict the build to", rather that "what jdks to build with"

it's easy now to completely remove the hardcoded methods, but it means changing the 'jdk' parameter from choice to string.

Reply:
commits pushed. please check.